internal/topology/variables/clusterclass_variable_validation.go

```diff
@@ -122,13 +150,30 @@ func validateClusterClassVariableMetadata(metadata clusterv1.ClusterClassVariabl
 var validVariableTypes = sets.Set[string]{}.Insert("object", "array", "string", "number", "integer", "boolean")
 
 // validateRootSchema validates the schema.
-func validateRootSchema(ctx context.Context, clusterClassVariable *clusterv1.ClusterClassVariable, fldPath *field.Path) field.ErrorList {
+func validateRootSchema(ctx context.Context, oldClusterClassVariables, clusterClassVariable *clusterv1.ClusterClassVariable, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 
-	apiExtensionsSchema, allErrs := convertToAPIExtensionsJSONSchemaProps(&clusterClassVariable.Schema.OpenAPIV3Schema, field.NewPath("schema"))
-	if len(allErrs) > 0 {
-		return field.ErrorList{field.InternalError(fldPath,
-			fmt.Errorf("failed to convert schema definition for variable %q; ClusterClass should be checked: %v", clusterClassVariable.Name, allErrs))} // TODO: consider if to add ClusterClass name
+	apiExtensionsSchema, validationErrors := convertToAPIExtensionsJSONSchemaProps(&clusterClassVariable.Schema.OpenAPIV3Schema, fldPath)
+	if len(validationErrors) > 0 {
+		for _, validationError := range validationErrors {
+			// Add context to the error message.
+			validationError.Detail = fmt.Sprintf("failed to convert schema definition for variable %q; ClusterClass should be checked: %s", clusterClassVariable.Name, validationError.Detail) // TODO: consider if to add ClusterClass name
```

Comment:
Should we resolve this todo or remove it before merging? (Same below)

Reply:
Nope, it's a copy & paste from a lot of other places (has been there since forever). Let's solve this independently